Build error on Android

```
/sdcard/dev/cpp/Bedrock++/lib/RakNet/src/FileList.cc:24:10: fatal error: 'sys/io.h' file
      not found
#include <sys/io.h>
         ^~~~~~~~~~
1 error generated.
make[2]: *** [lib/RakNet/CMakeFiles/RakNetStatic.dir/build.make:567: lib/RakNet/CMakeFiles/RakNetStatic.dir/src/FileList.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:88: lib/RakNet/CMakeFiles/RakNetStatic.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

This also happens with the original RakNet library, could probably do with some checks.

I also get errors when including the RakNet namespace. I assume i need to include every source file but i'm wondering how to do that without typing it all individually.

```
$ cmake --build .
[ 49%] Built target RakNetStatic
[ 98%] Built target RakNetShared
Scanning dependencies of target bedrock++
[ 99%] Building CXX object src/CMakeFiles/bedrock++.dir/Server.cpp.o
[ 99%] Building CXX object src/CMakeFiles/bedrock++.dir/util/ChatFormat.cpp.o
[100%] Linking CXX executable bedrock++
CMakeFiles/bedrock++.dir/Server.cpp.o: In function `Server::Start(int, char*, int, char*)':
/sdcard/dev/cpp/Bedrock++/src/Server.cpp:(.text+0x1c): undefined reference to `RakNet::RakPeerInterface::GetInstance()'
/sdcard/dev/cpp/Bedrock++/src/Server.cpp:(.text+0x40): undefined reference to `RakNet::SocketDescriptor::SocketDescriptor(unsigned short, char const*)'
CMakeFiles/bedrock++.dir/Server.cpp.o: In function `__cxx_global_var_init':
/sdcard/dev/cpp/Bedrock++/src/Server.cpp:(.text.startup+0x14): undefined reference to `RakNet::SystemAddress::SystemAddress()'
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/CMakeFiles/bedrock++.dir/build.make:147: src/bedrock++] Error 1
make[1]: *** [CMakeFiles/Makefile2:180: src/CMakeFiles/bedrock++.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/src/FileList.cc
+++ b/src/FileList.cc
@@ -14,13 +14,12 @@
 
 #include <stdio.h>  // RAKNET_DEBUG_PRINTF
 #include "RakAssert.h"
-#if defined(ANDROID)
-#include <asm/io.h>
-#elif defined(_WIN32) || defined(__CYGWIN__)
+
+#if defined(_WIN32) || defined(__CYGWIN__)
 #include <io.h>
 
 #elif !defined(__APPLE__) && !defined(__APPLE_CC__) && !defined(__PPC__) && \
-    !defined(__FreeBSD__) && !defined(__S3E__)
+    !defined(ANDROID) && !defined(__ANDROID__) && !defined(__FreeBSD__) && !defined(__S3E__)
 #include <sys/io.h>
 #endif
 

